ext/standard/dir.c

```diff
@@ -415,8 +415,8 @@ PHP_FUNCTION(glob)
 	ZEND_PARSE_PARAMETERS_END();
 
 	if (pattern_len >= MAXPATHLEN) {
-		php_error_docref(NULL, E_WARNING, "Pattern exceeds the maximum allowed length of %d characters", MAXPATHLEN);
-		RETURN_FALSE;
+		zend_argument_value_error(1, "must have a length less than %d bytes", MAXPATHLEN);
```

Comment:
I'm clearly not sure how to compose these messages :)

Reply:
And I'm also not exactly sure that we promote warnings related to maximum path length

Reply:
I'd be inclined not to promote them. I think that "path too long" and "path not found" should be considered as one error condition. It would be awkward if calling code had to check path length (I'm not even sure we export MAXPATHLEN).